Lib/test/test_binascii.py

```diff
@@ -114,6 +114,38 @@ def addnoise(line):
         # empty strings. TBD: shouldn't it raise an exception instead ?
         self.assertEqual(binascii.a2b_base64(self.type2test(fillers)), b'')
 
+    def test_base64_strict_mode(self):
+        # Test base64 with strict mode on
+        def assertExcessData(data):
+            with self.assertRaisesRegex(binascii.Error, r'(?i)Excess data'):
+                binascii.a2b_base64(self.type2test(data), strict_mode=True)
+
+        def assertNonBase64Data(data):
+            with self.assertRaisesRegex(binascii.Error, r'(?i)Only base64 data'):
+                binascii.a2b_base64(self.type2test(data), strict_mode=True)
+
+        def assertMalformedPadding(data):
+            with self.assertRaisesRegex(binascii.Error, r'(?i)Malformed padding'):
+                binascii.a2b_base64(self.type2test(data), strict_mode=True)
+
+        # Test excess data exceptions
```

Comment:
Add edge case tests for `b'='` through `b'==='`.

Reply:
Decided to disallow `=...` (leading padding), since it's not a valid base64 (afb95db).
Added tests in 3c5758b.